env0/resource_configuration_variable.go

```diff
@@ -237,6 +242,11 @@ func resourceConfigurationVariableUpdate(ctx context.Context, d *schema.Resource
 }
 
 func resourceConfigurationVariableDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// don't delete if soft delete is set
+	if softDelete, ok := d.GetOk("soft_delete"); ok && softDelete.(bool) {
```

Comment:
If you put a default you can just get the softDelete.
GetOk does not work properly with booleans...

Reply:
[eed48af](https://github.com/env0/terraform-provider-env0/pull/866/commits/eed48af5b5c5d63a6c376f7d7d746f992270db90)